server/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -1601,14 +1601,12 @@ public void testScoreReferenceInvalidNotNullPredicate() throws Exception {
 
 
     @Test
-    public void testRegexpMatchInvalidArg() throws Exception {
+    public void test_regex_match_on_non_string_columns_use_casts() throws Exception {
         var executor = SQLExecutor.builder(clusterService)
             .addTable(TableDefinitions.USER_TABLE_DEFINITION)
             .build();
-        expectedException.expect(UnsupportedOperationException.class);
-        expectedException.expectMessage("Unknown function: (doc.users.floats ~ 'foo')," +
-                                        " no overload found for matching argument types: (real, text).");
-        executor.analyze("select * from users where floats ~ 'foo'");
+        QueriedSelectRelation stmt = executor.analyze("select * from users where floats ~ 'foo'");
+        assertThat(stmt.where(), isSQL("(_cast(doc.users.floats, 'text') ~ 'foo')"));
```

Comment:
This would mean that the floats are casted to text and then matched against `foo`, right ?

Reply:
yep. Before the query failed with an error that there is no overload.
We could also keep it as strict but then I think we should remove the undefined overload as well. (and either special case it in the coersion, or not support it anymore)

@seut what do you think? ^

Reply:
I think its fine to add support for non-string types by casts.